Sources/GenericModule/View/ViewController.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  Copyright © 2021 Rosberry. All rights reserved.
+//
+
+import UIKit
+
+open class ViewController<ViewModel: GenericModule.ViewModel, ViewInput, Output>: UIViewController, View {
+    public var output: Output
+    public var viewModel: ViewModel
+
+    private var genericModuleViewOutput: ViewOutput? {
+        output as? ViewOutput
+    }
+
+    // MARK: - Lifecycle
+
+    public required init(viewModel: ViewModel, output: Output) {
+        self.viewModel = viewModel
+        self.output = output
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    public required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    open override func viewDidLoad() {
```

Comment:
As I remember we've discussed the case when you need to override viewDidLoad you need to call `super.viewDidLoad`, but this super call shouldn't call output's viewDidLoad. Do you remember which solution we pick?

Reply:
As I remember the main purpose of this `ViewController` is automatic presenter appearance methods call